appcelerator-archive/amp
# Pull images on function worker nodes on creation and remove images on function removal

## How to test
```
ampmake clean build
alias amp='amp -s localhost'
amp cluster create --tag=local
amp user signup --name user --email your@email.com --password password
amp user verify token
amp login --name user --password password
```

In another terminal:
```
. ./hack/ampenv
amps service logs -f amp_funcexec
```

Then back to the first terminal:
```
amp fn create banner bquenin/banner-functio
amp fn rm function-id
```

Check the logs of the second terminal for:
```
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | 2017/04/25 00:02:18 Function has been created: id:"10155a0303da7d4e76900c7d0e61526dea5d9a3f20b0f04699f1efa22be35055" name:"banner" image:"bquenin/banner-function" owner:<name:"user" > create_dt:1493078538 
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | latest: Pulling from bquenin/banner-function
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | 627beaf3eaaf: Already exists
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | 039235d19caa: Pulling fs layer
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | 039235d19caa: Download complete
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | 039235d19caa: Pull complete
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | Digest: sha256:3442dd6535c7528b7d34ca13946d5c60c0067a6ebbe4512656d2b0c326bcc787
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | Status: Downloaded newer image for bquenin/banner-function:latest
amp_funcexec.1.sypmicbs4mzj@5d4a78de5ad7    | 2017/04/25 00:02:31 Function has been deleted: id:"10155a0303da7d4e76900c7d0e61526dea5d9a3f20b0f04699f1efa22be35055" name:"banner" image:"bquenin/banner-function" owner:<name:"user" > create_dt:1493078538 
```
